### PR TITLE
Define the interface to save created swaps to the db

### PR DESCRIPTION
**Pull request by tcharding**
Users can create an ether/lightning swap via the REST API.  We would
like a way to save the data that the user provides into the database.
As a first step in doing so define an interface between the controller
and the database.

(The logic in this PR was originally done in: #2513)

Fixes: #2158

### CODE REVIEW

File: cnd/src/db.rs [hunk 1]
```diff
@@ -132,6 +133,25 @@ pub enum Error {
...
+/// Data required to create a swap.
+///
+/// 'create' a swap is defined as the process of initiating a swap within `cnd`.
+/// The data required to do so is assumed to have been negotiated between the
+/// two parties prior to each creating the swap.
+#[derive(Debug, Clone)]
+pub struct CreatedSwap<A, B> {
+    /// Node specific swap identifier.
+    pub swap_id: LocalSwapId,
+    /// The protocol used on the alpha ledger.
```
Comment (D4nte) [inline cnd/src/db.rs:145]: This also contains asset quantities, right?
Hence I am not sure `protocol` is the best description.
Maybe `parameters for the alpha ledger`?

File: cnd/src/swap_protocols/herc20.rs [hunk 2]
```diff
@@ -21,7 +21,19 @@ use crate::{ethereum::Bytes, timestamp::Timestamp};
 use blockchain_contracts::ethereum::rfc003::Erc20Htlc;
 pub use connector_impls::*;
 
-/// Resolves when said event has occured.
+/// Htlc ERC20 Token atomic swap protocol.
+
+/// Data required to create a swap that involves an ERC20 token.
```
Comment (thomaseizinger) [inline cnd/src/swap_protocols/herc20.rs:26]: Not fully true, we are missing the other person's identity :)

Comment (tcharding) [reply cnd/src/swap_protocols/herc20.rs:26]: We don't require the other parties identity to create a swap, right?  We require that to 'finalize' a swap i.e., we learn it during the communication protocols.

File: cnd/src/db.rs (hunk 1 above)
Comment (tcharding) [reply cnd/src/db.rs:145]: Re-named as suggested, thanks.

File: cnd/src/swap_protocols/herc20.rs (hunk 2 above)
Comment (tcharding) [reply cnd/src/swap_protocols/herc20.rs:26]: FTR I defined 'create a swap' in `db.rs` on the `CreatedSwap` struct.

Comment (thomaseizinger) [reply cnd/src/swap_protocols/herc20.rs:26]: Depends on what you mean by "create a swap". If that means create the HTLCs / watch for them, then we do require the identites.

Comment (tcharding) [reply cnd/src/swap_protocols/herc20.rs:26]: I defined 'create a swap' in line with the identifiers we use `CreatedSwap` i.e., initiate the swap in cnd, post the swap on the REST API.

Comment (tcharding) [reply cnd/src/swap_protocols/herc20.rs:26]: oh, this might be a circular argument.  I thought we had this concept 'created' swap and 'finalized' swap.

Comment (thomaseizinger) [reply cnd/src/swap_protocols/herc20.rs:26]: > I thought we had this concept 'created' swap and 'finalized' swap.

Unfortunately, I don't think that has settled yet as we can see from this discussion :see_no_evil:

Comment (tcharding) [reply cnd/src/swap_protocols/herc20.rs:26]: Ok, so I'm not sure what the action to carry out is after this thread of conversation, possible options:
- Remove all the comments on the various `CreatedSwap` structs?
- Rename all the `CreatedSwap` structs to something else?

I can do either of these but personally I couldn't come up with anything better than whats here, a not-perfect name with documentation explaining what the name means.

FTR, this patch also adds (`db.rs`)
```
/// Data required to create a swap.
///
/// 'create' a swap is defined as the process of initiating a swap within `cnd`.
/// The data required to do so is assumed to have been negotiated between the
/// two parties prior to each creating the swap.
#[derive(Debug, Clone)]
pub struct CreatedSwap<A, B> {
```

**Comment by tcharding** [comment]
Hey @thomaseizinger, is there an issue with this PR that I've overlooked.  Its been sitting for 5 days with no comments.  @rishflab and @da-kami I have you requested as reviewers also please.

**Comment by thomaseizinger** [comment]
> Hey @thomaseizinger, is there an issue with this PR that I've overlooked. Its been sitting for 5 days with no comments. @rishflab and @da-kami I have you requested as reviewers also please.

Sorry, was just quite busy. Will try to look into it today!

**Pull request merged**